ledger/participant-integration-api/src/test/suite/scala/db/migration/translation/apiserver/services/admin/ApiConfigManagementServiceSpec.scala

```diff
@@ -3,51 +3,200 @@
+import scala.util.{Failure, Success}
 
 class ApiConfigManagementServiceSpec
     extends AsyncWordSpec
-    with TelemetrySpecBase
-    with MockitoSugar
     with Matchers
+    with Inside
+    with MockitoSugar
     with ArgumentMatchersSugar
+    with TelemetrySpecBase
     with AkkaBeforeAndAfterAll {
 
-  import ApiConfigManagementServiceSpec._
-
   private implicit val loggingContext: LoggingContext = LoggingContext.ForTesting
 
   "ApiConfigManagementService" should {
-    "propagate trace context" in {
-      val mockReadBackend = mockIndexConfigManagementService()
+    "get the time model" in {
+      val indexedTimeModel = LedgerTimeModel(
+        avgTransactionLatency = Duration.ofMinutes(5),
+        minSkew = Duration.ofMinutes(3),
+        maxSkew = Duration.ofMinutes(2),
+      ).get
+      val expectedTimeModel = TimeModel.of(
+        avgTransactionLatency = Some(DurationProto.of(5 * 60, 0)),
+        minSkew = Some(DurationProto.of(3 * 60, 0)),
+        maxSkew = Some(DurationProto.of(2 * 60, 0)),
+      )
 
+      val writeService = mock[state.WriteConfigService]
       val apiConfigManagementService = ApiConfigManagementService.createApiService(
-        mockReadBackend,
+        new FakeCurrentIndexConfigManagementService(
+          LedgerOffset.Absolute(Ref.LedgerString.assertFromString("0")),
+          Configuration(aConfigurationGeneration, indexedTimeModel, Duration.ZERO),
+        ),
+        writeService,
+        TimeProvider.UTC,
+      )
+
+      apiConfigManagementService.getTimeModel(GetTimeModelRequest.defaultInstance).map { response =>
+        response.timeModel should be(Some(expectedTimeModel))
+        verifyZeroInteractions(writeService)
+        succeed
+      }
+    }
+
+    "return a `NOT_FOUND` error if a time model is not found" in {
+      val writeService = mock[state.WriteConfigService]
+      val apiConfigManagementService = ApiConfigManagementService.createApiService(
+        EmptyIndexConfigManagementService,
+        writeService,
+        TimeProvider.UTC,
+      )
+
+      apiConfigManagementService
+        .getTimeModel(GetTimeModelRequest.defaultInstance)
+        .transform(Success.apply)
+        .map { response =>
+          response should matchPattern { case Failure(GrpcException(GrpcStatus.NOT_FOUND(), _)) => }
+        }
+    }
+
+    "set a new time model" in {
+      val maximumDeduplicationTime = Duration.ofHours(6)
+      val initialGeneration = 2L
+      val initialConfiguration = Configuration(
+        generation = initialGeneration,
+        timeModel = LedgerTimeModel(
+          avgTransactionLatency = Duration.ofMinutes(1),
+          minSkew = Duration.ofMinutes(2),
+          maxSkew = Duration.ofMinutes(3),
+        ).get,
+        maxDeduplicationTime = maximumDeduplicationTime,
+      )
+      val expectedGeneration = 3L
+      val expectedConfiguration = Configuration(
+        generation = expectedGeneration,
+        timeModel = LedgerTimeModel(
```

Comment:
For readability, you might want to move the two time models required for these tests into a helper variable (e.g., `aTimeModel` and `aNewerTimeModel`).